src/Compilers/Core/CommandLine/CompilerServerLogger.cs

```diff
@@ -21,130 +96,79 @@ namespace Microsoft.CodeAnalysis.CommandLine
                 {
                     // If the environment variable contains the path of a currently existing directory,
                     // then use a process-specific name for the log file and put it in that directory.
                     // Otherwise, assume that the environment variable specifies the name of the log file.
                     if (Directory.Exists(loggingFileName))
                     {
-                        loggingFileName = Path.Combine(loggingFileName, $"server.{GetCurrentProcessId()}.log");
+                        loggingFileName = Path.Combine(loggingFileName, $"server.{_processId}.log");
                     }
 
                     // Open allowing sharing. We allow multiple processes to log to the same file, so we use share mode to allow that.
-                    s_loggingStream = new FileStream(loggingFileName, FileMode.OpenOrCreate, FileAccess.Write, FileShare.ReadWrite);
+                    _loggingStream = new FileStream(loggingFileName, FileMode.OpenOrCreate, FileAccess.Write, FileShare.ReadWrite);
                 }
             }
             catch (Exception e)
             {
-                LogException(e, "Failed to create logging stream");
-            }
-        }
-
-        /// <summary>
-        /// Set the logging prefix that describes our role.
-        /// Typically a 3-letter abbreviation. If logging happens before this, it's logged with "---".
-        /// </summary>
-        public static void Initialize(string outputPrefix)
-        {
-            s_prefix = outputPrefix;
-        }
-
-        /// <summary>
-        /// Log an exception. Also logs information about inner exceptions.
-        /// </summary>
-        public static void LogException(Exception exception, string reason)
-        {
-            if (s_loggingStream != null)
-            {
-                LogError("'{0}' '{1}' occurred during '{2}'. Stack trace:\r\n{3}", exception.GetType().Name, exception.Message, reason, exception.StackTrace);
-
-                int innerExceptionLevel = 0;
-
-                Exception? e = exception.InnerException;
-                while (e != null)
-                {
-                    Log("Inner exception[{0}] '{1}' '{2}'. Stack trace: \r\n{3}", innerExceptionLevel, e.GetType().Name, e.Message, e.StackTrace);
-                    e = e.InnerException;
-                    innerExceptionLevel += 1;
-                }
+                Debug.Assert(false, e.Message);
             }
         }
 
-        /// <summary>
-        /// Log a line of text to the logging file, with string.Format arguments.
-        /// </summary>
-        public static void Log(string format, params object?[] arguments)
+        public void Dispose()
         {
-            if (s_loggingStream != null)
-            {
-                Log(string.Format(format, arguments));
-            }
+            _loggingStream?.Dispose();
```

Comment:
Do we need a finalizer pattern to ensure that dispose is run on finalization?

Reply:
No. The underlying `Stream` here already has a finalizer and hence will take care of this for us.